src/scripting/lua_map.cc

```diff
@@ -838,8 +838,31 @@ int upcasted_map_object_descr_to_lua(lua_State* L, const Widelands::MapObjectDes
 			return CAST_TO_LUA(Widelands::WorkerDescr, LuaWorkerDescription);
 		case Widelands::MapObjectType::SOLDIER:
 			return CAST_TO_LUA(Widelands::SoldierDescr, LuaSoldierDescription);
+		case Widelands::MapObjectType::SHIP:
+			return CAST_TO_LUA(Widelands::ShipDescr, LuaShipDescription);
 		case Widelands::MapObjectType::IMMOVABLE:
 			return CAST_TO_LUA(Widelands::ImmovableDescr, LuaImmovableDescription);
+		case Widelands::MapObjectType::RESOURCE:
+			return CAST_TO_LUA(Widelands::ResourceDescription, LuaResourceDescription);
+		case Widelands::MapObjectType::TERRAIN:
+			return CAST_TO_LUA(Widelands::TerrainDescription, LuaTerrainDescription);
+		/*
```

Comment:
Oh one thing left, please either uncomment or delete this

Reply:
I uncommented it.